panel/src/components/Forms/Input/TextareaInput.vue

```diff
@@ -291,36 +299,44 @@ export default {
 				title: selection
 			};
 		},
-		prepend(text) {
-			this.insert(text + " " + this.selection());
+		async prepend(text) {
+			return this.insert(text + " " + this.selection());
 		},
+		/**
+		 * @deprecated 5.0.0 Use `setSelectionRange` instead
+		 */
 		restoreSelectionCallback() {
 			// store selection
-			const start = this.$refs.input.selectionStart;
-			const end = this.$refs.input.selectionEnd;
+			const selectionRange = this.selectionRange();
 
 			// restore selection as `insert` method
 			// depends on it
-			return (callback) => {
-				setTimeout(() => {
-					this.$refs.input.setSelectionRange(start, end);
+			return async (callback) => {
+				await this.setSelectionRange(selectionRange);
 
-					if (callback) {
-						callback();
-					}
-				});
+				if (callback) {
+					callback();
+				}
 			};
 		},
 		select() {
 			this.$refs.select();
 		},
 		selection() {
-			return this.$refs.input.value.substring(
-				this.$refs.input.selectionStart,
-				this.$refs.input.selectionEnd
-			);
+			const { start, end } = this.selectionRange();
+			return this.$refs.input.value.substring(start, end);
+		},
+		selectionRange() {
+			return {
+				start: this.$refs.input.selectionStart,
+				end: this.$refs.input.selectionEnd
+			};
+		},
+		async setSelectionRange({ start, end }) {
+			this.$refs.input.setSelectionRange(start, end);
+			return this.$nextTick();
```

Comment:
```suggestion
			await this.$nextTick();
```

`Suggestion:` I feel we do not actually want to return the promise - this method probably should run as void, shouldn't it? But await the next tick before the method resolves.